app/Console/Commands/ImportBusinessesSpreadsheet.php

```diff
@@ -0,0 +1,219 @@
+                        'warranty_offered', 'warranty_details', 'publish'
+                    ];
+
+                    for ($i = 0; $i < count($fields); $i++) {
+                        $field = $fields[$i];
+                        $$field = trim($sheet->getCellByColumnAndRow($i + 1, $row)->getValue());
+                    }
+
+                    $key = $name . $address;
+
+                    if (array_key_exists($key, $uniques)) {
+                        $this->error("Duplicate business $name, $address");
+                    } else if (strlen($review_url) > 1000) {
+                        $this->error("$name has overlong review URL $review_url");
+                    } else {
+                        $uniques[$key] = TRUE;
+
+                        // Tweak the URL.
+                        if ($website) {
+                            if (stripos($website, 'http') === FALSE) {
+                                $website = "https://$website";
+                            }
+
+                            $website = preg_replace('/\/#$/', '', $website);
+                            $website = preg_replace('/\/$/', '', $website);
+                        }
+
+                        // The items repaired are the next columns, until we hit an empty one.
+                        $products = [];
+                        $col = count($fields) + 1;
+
+                        while ($sheet->getCellByColumnAndRow($col, $row)->getValue()) {
+                            $prod = $sheet->getCellByColumnAndRow($col++, $row)->getValue();
+
+                            if (strcmp(strtolower($prod),  'games console') === 0) {
+                                # Special case - list the individual brands.
+                                $products[] = Category::NINTENDO_CONSOLE;
+                                $products[] = Category::XBOX_CONSOLE;
+                                $products[] = Category::PLAYSTATION_CONSOLE;
+                            } else {
+                                $products[] = $prod;
+                            }
+                        }
+
+                        $products = array_unique($products);
+
+                        $business = new Business();
+                        $business->setName($name);
+                        $business->setUpdatedAt(new \DateTime("now"));
+
+                        $point = null;
+
+                        try {
+                            $point = $geocoder->geocode("$address,$city,$borough,$postcode");
+
+                            if ($point && ($point->getLatitude() < $swlat || $point->getLatitude() > $nelat || $point->getLongitude() < $swlng || $point->getLongitude() > $nelng)) {
+                                $this->error("$name address $address geocodes to invalid lat/lng " . $point->getLatitude() . "," . $point->getLongitude());
+                                $point = null;
```

Comment:
To check, are we saying here that if the address doesn't geocode, we are ignoring the business and logging it as a problem?  Presumably to then send to REW and ask them to fix.

Reply:
Yes; will comment to that effect.